Xamarin.Forms.Platform.WinRT/CellControl.cs

```diff
@@ -215,32 +215,34 @@ void SetCell(object newContext)
 			{
 				bool isGroupHeader = IsGroupHeader;
 				DataTemplate template = isGroupHeader ? lv.GroupHeaderTemplate : lv.ItemTemplate;
+				object bindingContext = newContext;
 
 				if (template is DataTemplateSelector)
 				{
-					template = ((DataTemplateSelector)template).SelectTemplate(newContext, lv);
+					template = ((DataTemplateSelector)template).SelectTemplate(bindingContext, lv);
 				}
 
 				if (template != null)
 				{
 					cell = template.CreateContent() as Cell;
-					cell.BindingContext = newContext;
 				}
 				else
 				{
 					IListViewController listViewController = lv;
-					var defaultContext = newContext;
 
 					if (isGroupHeader)
-						defaultContext = listViewController.GetDisplayTextFromGroup(newContext);
+						bindingContext = listViewController.GetDisplayTextFromGroup(bindingContext);
 
-					cell = listViewController.CreateDefaultCell(defaultContext);
+					cell = listViewController.CreateDefaultCell(bindingContext);
 				}
 
 				// A TableView cell should already have its parent,
 				// but we need to set the parent for a ListView cell.
 				cell.Parent = lv;
```

Comment:
Hi @samhouts,

Here is a problem, that causes the bug https://bugzilla.xamarin.com/show_bug.cgi?id=52841.
When you set `cell.Parent` and a cell does not have a `BindingContext`, then parent's context is inherited to each cell. Which causes custom checks (casts) to fail in `OnBindingContextChanged` for each custom cell.

Then when you set `BindableObject.SetInheritedBindingContext`, a proper context is set, that is correct.